packages/@react-aria/interactions/src/utils.ts

```diff
@@ -111,10 +116,19 @@ export function useSyntheticBlurEvent<Target = Element>(onBlur: (e: ReactFocusEv
         }
       };
 
+      if (stateRef.current.target && stateRef.current.listener) {
+        stateRef.current.target.removeEventListener('focusout', stateRef.current.listener);
```

Comment:
Why do some parts of this function use `target` directly and others use `stateRef.current.target`? Why does the target need to be in the ref at all?